### ISSUE TITLE
MGLGeoJSONSource does not handle exceptions from GeoJSON parsing

### ISSUE THREAD

**Issue by jfirebaugh** [mapbox/mapbox-gl-native#6987]
The following code crashes the application with

> libc++abi.dylib: terminating with uncaught exception of type std::runtime_error: Feature must have a geometry property

It should instead raise the error using appropriate cocoa error reporting mechanism.

```
    MGLGeoJSONSource *geojsonSource = [[MGLGeoJSONSource alloc]
        initWithIdentifier:@"geojson"
               geoJSONData:[@"{\"type\": \"Feature\", \"feature\": {\"type\": \"Point\", \"coordinates\": [0, 0]}}" dataUsingEncoding:NSUTF8StringEncoding]
                   options:nil];
```

**Comment by nitrag** [comment]
In the event that I query several hundred features, I wouldn't want it to fail just because one feature was corrupted and didn't have a geometry. Rather, it should throw a warning and ignore/skip that particular feature.

**Comment by 1ec5** [comment]
#7334 renamed MGLGeoJSONSource to MGLShapeSource. #7377 removes the initializer in question, in favor of a general-purpose deserialization method (for #3630). This method handles errors gracefully via an NSError out parameter, which is represented as the `throws` keyword in Swift.

**Comment by 1ec5** [comment]
Fixed in #7377 on the iOS SDK v3.4.0 release branch.
